lib/src/desktop/upload_isolate.dart

```diff
@@ -50,7 +50,7 @@ Future<TaskStatus> binaryUpload(
     request.contentLength = fileSize;
     request.headers['Content-Type'] = task.mimeType;
     request.headers['Content-Disposition'] =
-        'attachment; filename="${task.filename}"';
+        'attachment; filename="${Uri.encodeComponent(task.filename)}"';
```

Comment:
Per the `browserEncode` comment: 
```
// http://tools.ietf.org/html/rfc2388 mandates some complex encodings for
// field names and file names, but in practice user agents seem not to
// follow this at all. Instead, they URL-encode `\r`, `\n`, and `\r\n` as
// `\r\n`; URL-encode `"`; and do nothing else (even for `%` or non-ASCII
// characters). We follow their behavior.
```
This behavior and code is copied from the dart:http package, so I would like us to  follow that same convention and not "properly" url-encode the filename. That means we should call `browserEncode` here, and not `Uri.encodeComponent`.

Reply:
Yeah, I wasn't sure about conformity to standards, I was hoping you would clear that up. It did, however, fix my issue with encoding errors.

Here is the error when trying to upload a file with `browserEncode` and the filename contains unicode characters:

`FormatException: Invalid HTTP header field value: "attachment; filename=\"testiņš_maziņš.txt\"" (at character 28)` (character 28 being the first unicode character 'ņ')

When `encodeComponent` is added (with or without `browserEncode`), the file uploads without any issue.